apps/studio/components/interfaces/DiskManagement/DiskManagement.utils.ts

```diff
@@ -109,3 +147,195 @@ export const calculateThroughputPrice = ({
+      connections_pooler: INSTANCE_NANO_SPECS.connections_pooler,
+    } as ProjectAddonVariantMeta,
+  })
+
+  return computeOptions
+}
+
+export const calculateMaxIopsAllowedForDiskSizeWithGp3 = (totalSize: number) => {
+  if (!totalSize) throw Error('Total size is required')
+  return Math.min(3000 * totalSize, 16000)
+}
+
+export const calculateDiskSizeRequiredForIopsWithGp3 = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.max(1, Math.ceil(iops / 500))
+}
+
+export const calculateMaxIopsAllowedForDiskSizeWithio2 = (totalSize: number) => {
+  if (!totalSize) throw Error('Total size is required')
+  return Math.min(500 * totalSize, 256000)
+}
+
+export const calculateDiskSizeRequiredForIopsWithIo2 = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.max(4, Math.ceil(iops / 1000))
+}
+
+export const calculateMaxThroughput = (iops: number) => {
+  if (!iops) throw Error('IOPS is required')
+  return Math.min(0.256 * iops, 1000)
+}
+
+export const calculateIopsRequiredForThroughput = (throughput: number) => {
+  if (!throughput) throw Error('Throughput is required')
+  return Math.max(125, Math.ceil(throughput / 0.256))
+}
+
+export const calculateMaxIopsAllowedForComputeSize = (computeSize: string): number => {
+  if (!computeSize) throw Error('Compute size is required')
+  return COMPUTE_BASELINE_IOPS[computeSize as keyof typeof COMPUTE_BASELINE_IOPS] || 0
+}
+
+export const calculateComputeSizeRequiredForIops = (
+  iops: number
+): ComputeInstanceAddonVariantId => {
+  if (iops === undefined || iops === null) {
+    throw new Error('IOPS is required')
+  }
+
+  const computeSizes = Object.entries(COMPUTE_BASELINE_IOPS).sort((a, b) => a[1] - b[1])
+  for (const [size, baselineIops] of computeSizes) {
+    if (iops <= baselineIops) {
+      return size as ComputeInstanceAddonVariantId
+    }
+  }
+
+  // fallback to largest compute size
+  // this should never happen though :-/
+  return computeSizes[computeSizes.length - 1][0] as ComputeInstanceAddonVariantId
+}
+
+export const calculateDiskSizeRequiredForIops = (provisionedIOPS: number): number => {
+  if (provisionedIOPS === undefined || provisionedIOPS === null) {
+    throw new Error('Provisioned IOPS is required')
+  }
+
+  if (typeof provisionedIOPS !== 'number' || isNaN(provisionedIOPS) || provisionedIOPS < 0) {
+    throw new Error('Provisioned IOPS must be a non-negative number')
+  }
+
+  if (provisionedIOPS > 256000) {
+    throw new Error('Maximum allowed IOPS is 256000')
+  }
+
+  return Math.max(1, Math.ceil(provisionedIOPS / 1000))
+}
+
+export const formatComputeName = (compute: string) => {
+  return compute.toUpperCase().replace('CI_', '')
+}
+
+export const mapComputeSizeNameToAddonVariantId = (
+  computeSize: ProjectDetail['infra_compute_size']
+): ComputeInstanceAddonVariantId => {
+  if (!computeSize) throw new Error('Compute size is required')
```

Comment:
this check seems unnecessary too - the props are already declared as non optional